match2/commons/match_group_display.lua

```diff
@@ -5,17 +5,78 @@ function MatchGroupDisplay.bracket(frame)
 	return MatchGroupDisplay.luaBracket(frame, args)
 end
 
-function MatchGroupDisplay.luaBracket(frame, args)
+function MatchGroupDisplay.customBracket(frame, args, matches)
+	if not args then
+		args = require('Module:Arguments').getArgs(frame)
+	end
+	if not matches or #matches == 0 then
+		args[1] = args.id or args[1] or ''
+
+		local MatchGroupUtil = require('Module:MatchGroup/Util')
+		matches = MatchGroupUtil.fetchMatches(args[1])
+		if #matches == 0 then
+			error('No Data found for bracketId=' .. args[1])
+		end
+	end
+
+	if (args.title or '') ~= '' then
+		matches[1].bracketData.title = args.title
+	end
+
+	for ind, match in ipairs(matches) do
+		local matchId = string.gsub(match.matchId, args[1] .. '_', '')
+		local round, matchInRound = string.match(matchId, '^R(%d+)%-M(%d+)$')
```

Comment:
You don't have to do this, but just a tip:

This regex takes a second to grasp, so this code would be improved by extracting this part into a function, then it would just be e.g. `getRoundFromMatchId` and readers don't need to spend energy to grasp what is going on anymore